trie-db can not be used as crate dependencies after 0.19.2

### How to reproduce
- create a new project by cargo, ie `cargo init --bin <project_name>`
- add `trie-db` as a part of dependencies in `Cargo.toml`
- `cargo build` will fail in these versions, `0.20`, `0.21` with the following error
- It can be build with version `0.19.2`
```
error[E0599]: no function or associated item named `new` found for struct `hashbrown::set::HashSet<_, _>` in the current scope
   --> /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/trie-db-0.21.0/src/triedbmut.rs:442:24
    |
442 |             death_row: HashSet::new(),
    |                                 ^^^

error[E0599]: no function or associated item named `new` found for struct `hashbrown::set::HashSet<_, _>` in the current scope
   --> /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/trie-db-0.21.0/src/triedbmut.rs:463:24
    |
463 |             death_row: HashSet::new(),
    |                                 ^^^

error[E0599]: no method named `insert` found for struct `hashbrown::set::HashSet<(<<L as TrieLayout>::Hash as hash_db::Hasher>::Out, (smallvec::SmallVec<[u8; 36]>, std::option::Option<u8>))>` in the current scope
   --> /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/trie-db-0.21.0/src/triedbmut.rs:519:21
    |
519 |                     self.death_row.insert((hash, current_key.left_owned()));
    |                                    ^^^^^^
    |
   ::: /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.6.3/src/map.rs:17:1
    |
17  | pub enum DefaultHashBuilder {}
    | --------------------------- doesn't satisfy `_: std::hash::BuildHasher`
    |
    = note: the method `insert` exists but the following trait bounds were not satisfied:
            `hashbrown::map::DefaultHashBuilder: std::hash::BuildHasher`

error[E0599]: no method named `insert` found for struct `hashbrown::set::HashSet<(<<L as TrieLayout>::Hash as hash_db::Hasher>::Out, (smallvec::SmallVec<[u8; 36]>, std::option::Option<u8>))>` in the current scope
   --> /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/trie-db-0.21.0/src/triedbmut.rs:523:21
    |
523 |                     self.death_row.insert((hash, current_key.left_owned()));
    |                                    ^^^^^^
    |
   ::: /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.6.3/src/map.rs:17:1
    |
17  | pub enum DefaultHashBuilder {}
    | --------------------------- doesn't satisfy `_: std::hash::BuildHasher`
    |
    = note: the method `insert` exists but the following trait bounds were not satisfied:
            `hashbrown::map::DefaultHashBuilder: std::hash::BuildHasher`

error[E0599]: no method named `insert` found for struct `hashbrown::set::HashSet<(<<L as TrieLayout>::Hash as hash_db::Hasher>::Out, (smallvec::SmallVec<[u8; 36]>, std::option::Option<u8>))>` in the current scope
    --> /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/trie-db-0.21.0/src/triedbmut.rs:1269:24
     |
1269 |                                 self.death_row.insert((
     |                                                ^^^^^^

     |
    ::: /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.6.3/src/map.rs:17:1
     |
17   | pub enum DefaultHashBuilder {}
     | --------------------------- doesn't satisfy `_: std::hash::BuildHasher`
     |
     = note: the method `insert` exists but the following trait bounds were not satisfied:
             `hashbrown::map::DefaultHashBuilder: std::hash::BuildHasher`

error[E0599]: no method named `insert` found for struct `hashbrown::set::HashSet<(<<L as TrieLayout>::Hash as hash_db::Hasher>::Out, (smallvec::SmallVec<[u8; 36]>, std::option::Option<u8>))>` in the current scope
    --> /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/trie-db-0.21.0/src/triedbmut.rs:1353:23
     |
1353 |                             self.death_row.insert(
     |                                            ^^^^^^
     |
    ::: /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.6.3/src/map.rs:17:1
     |
17   | pub enum DefaultHashBuilder {}
     | --------------------------- doesn't satisfy `_: std::hash::BuildHasher`
     |
     = note: the method `insert` exists but the following trait bounds were not satisfied:
             `hashbrown::map::DefaultHashBuilder: std::hash::BuildHasher`

error[E0599]: no method named `insert` found for struct `hashbrown::set::HashSet<(<<L as TrieLayout>::Hash as hash_db::Hasher>::Out, (smallvec::SmallVec<[u8; 36]>, std::option::Option<u8>))>` in the current scope
    --> /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/trie-db-0.21.0/src/triedbmut.rs:1372:23
     |
1372 |                             self.death_row.insert((hash, (child_prefix.0[..].into(), child_prefix.1)));
     |                                            ^^^^^^
     |
    ::: /home/yanganto/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.6.3/src/map.rs:17:1
     |
17   | pub enum DefaultHashBuilder {}
     | --------------------------- doesn't satisfy `_: std::hash::BuildHasher`
     |
     = note: the method `insert` exists but the following trait bounds were not satisfied:
             `hashbrown::map::DefaultHashBuilder: std::hash::BuildHasher`

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0599`.
error: could not compile `trie-db`.
```

--- a/trie-db/Cargo.toml
+++ b/trie-db/Cargo.toml
@@ -11,7 +11,7 @@ edition = "2018"
 log = "0.4"
 smallvec = "1.0.0"
 hash-db = { path = "../hash-db", default-features = false, version = "0.15.2"}
-hashbrown = { version = "0.8.0", default-features = false }
+hashbrown = { version = "0.8.0", default-features = false, features = ["ahash"] }
 rustc-hex = { version = "2.1.0", default-features = false, optional = true }
 
 [dev-dependencies]

